backend/TraderX/src/main/java/cmpe451/group6/rest/transaction/service/TransactionService.java

```diff
@@ -0,0 +1,200 @@
+package cmpe451.group6.rest.transaction.service;
+
+import cmpe451.group6.authorization.exception.CustomException;
+import cmpe451.group6.rest.asset.model.Asset;
+import cmpe451.group6.rest.asset.repository.AssetRepository;
+import cmpe451.group6.rest.equipment.configuration.EquipmentConfig;
+import cmpe451.group6.rest.equipment.model.Equipment;
+import cmpe451.group6.rest.equipment.repository.EquipmentRepository;
+import cmpe451.group6.rest.transaction.dto.TransactionDTO;
+import cmpe451.group6.rest.transaction.model.Transaction;
+import cmpe451.group6.rest.transaction.model.TransactionType;
+import cmpe451.group6.rest.transaction.repository.TransactionRepository;
+import cmpe451.group6.authorization.model.User;
+import cmpe451.group6.authorization.repository.UserRepository;
+import cmpe451.group6.rest.follow.model.FollowStatus;
+import cmpe451.group6.rest.follow.repository.FollowRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Date;
+
+@Service
+public class TransactionService {
+
+    @Autowired
+    private TransactionRepository TransactionRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private FollowRepository followRepository;
+
+    @Autowired
+    private EquipmentRepository equipmentRepository;
+
+    @Autowired
+    private AssetRepository assetRepository;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+
```

Comment:
We need to consider funcitonalities of the functions here, I think we dont need functions that lists all users' activities; instead we need to focus on users one by one

Reply:
I see, as an extra functionality we can have such a function. Thx!